sympy/core/numbers.py

```diff
@@ -3456,7 +3457,7 @@ def approximation(self, number_cls):
 
     def _eval_evalf(self, prec):
         return Float._new(self._as_mpf_val(prec), prec)
-
+#ping
     def __eq__(self, other):
         try:
             other = _sympify(other)
```

Comment:
Again, only convert when appropriate. I also don't think the last if statement `if other.is_Number and self.is_irrational:` does anything, since it returns `False` regardless of whether the if statement is true or false